src/app/components/turnos/dashboard/demandaInsatisfecha.ts

```diff
@@ -53,7 +53,7 @@ export class demandaInsatisfechaComponent {
 
     guardar() {
         if (this.motivo && this.tipoPrestacion) {
-            if (this.serviceListaEspera.guardar(this.paciente, this.tipoPrestacion, this.estado, this.profesional, this.organizacion, this.motivo.nombre, this.origen)) {
+            if (this.listaEsperaService.save(this.paciente, this.tipoPrestacion, this.estado, this.profesional, this.organizacion, this.motivo.nombre, this.origen)) {
```

Comment:
El problema de hacer esto es que la respuesta del save es asincrona, es decir, lo que retorna es una promesa y recien a la vuelta de la api se obtiene el valor. Por lo tanto el if en primera instancia siempre va a dar true.
Lo correcto sería eliminar el if y dejar solo el llamado y, a la vuelta del subscribe, segun sea exito o error se muestra el toast correspondiente:
```suggestion
            this.listaEsperaService.save(this.paciente, this.tipoPrestacion, this.estado, this.profesional, this.organizacion, this.motivo.nombre, this.origen).subscribe(respuesta => {
            complete: () => {
                *** exito ***
            },
            error: () => {
                *** error ***
            }
    })
```

Reply:
Hecho